consumer/consumererror/README.md

```diff
@@ -0,0 +1,117 @@
+# Consumer errors
+
+This package contains error types that should be returned by a consumer when an
+error occurs while processing telemetry. The error types included in this
+package provide functionality for communicating details about
+the error for use upstream in the pipeline. Ideally the error returned by a
+component in its `consume` function should be from this package.
+
+## Use cases
+
+**Retry logic**: Errors should be allowed to embed rejected telemetry items to
+be retried by the consumer.
+
+**Indicating partial success**: Errors can indicate that not all items were
+accepted, for example as in an OTLP partial success message.
+
+**Communicating network error codes**: Errors should allow embedding information
+necessary for the Collector to act as a proxy for a backend, i.e. relay a status
+code returned from a backend in a response to a system upstream from the
+Collector.
+
+## Creating Errors
+
+Errors can be created through one of the three constructors provided:
+
+- `consumererror.New[Signal]` for retryable errors.
+- `consumererror.NewPartial` for errors where only some of the items failed.
+- `consumererror.NewHTTP` for errors resulting from an HTTP call with an error status code.
+- `consumererror.NewGRPC` for errors resulting from a gRPC call with an error status code.
+- `consumererror.NewComponent` for metadata around the component that produced the error.
+
+Only retryable errors will be retried, all other errors are considered permanent
+and will not be retried.
```

Comment:
We cannot assume this behavior. Probably "should" or "must" would be better fit here?
```suggestion
and should not be retried.
```